src/librustc_trans/mir/constant.rs

```diff
@@ -955,6 +952,55 @@ pub fn const_scalar_checked_binop<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
     }
 }
 
+unsafe fn const_cast_from_float(operand: &Const, signed: bool, int_ty: Type) -> ValueRef {
+    let llval = operand.llval;
+    // Note: this breaks if addresses can be turned into integers (is that possible?)
```

Comment:
This is valid rust code, isn't it?

```
println!("{}", {let w: * const u32 = {let v = 0u32; &v}; w} as usize);
```

Reply:
Yes, but this is constant evaluation and pointer->integer casts in constant expression [do not compile](https://play.rust-lang.org/?gist=6fee9754dcf18aa52ff07a2d477a061a&version=nightly):

```
static Y: i32 = 0;
static Z: usize = &Y as usize;
```